owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt

```diff
@@ -305,115 +309,105 @@ abstract class DrawerActivity : ToolbarActivity() {
+                            text = String.format(
+                                getString(R.string.drawer_quota),
+                                DisplayUtils.bytesToHumanReadable(userQuota.used, context, true),
+                                DisplayUtils.bytesToHumanReadable(userQuota.getTotal(), context, true),
+                                userQuota.getRelative()
+                            )
+                        }
+                        getAccountQuotaStatusText()?.apply {
+                            visibility = View.VISIBLE
+                            text = getString(R.string.drawer_nearing_quota)
+                        }
+                    } else if (userQuota.state == UserQuotaState.CRITICAL || userQuota.state == UserQuotaState.EXCEEDED) { // Critical storage. Value over 90%
+                        getAccountQuotaBar()?.apply {
+                            isVisible = true
+                            progress = userQuota.getRelative().toInt()
+                            progressTintList = ColorStateList.valueOf(resources.getColor(R.color.quota_exceeded))
+                        }
+                        getAccountQuotaText()?.apply {
+                            text = String.format(
+                                getString(R.string.drawer_quota),
+                                DisplayUtils.bytesToHumanReadable(userQuota.used, context, true),
+                                DisplayUtils.bytesToHumanReadable(userQuota.getTotal(), context, true),
+                                userQuota.getRelative()
+                            )
+                        }
+                        getAccountQuotaStatusText()?.apply {
+                            visibility = View.VISIBLE
+                            text = getString(R.string.drawer_critical_quota)
+                        }
+                    } else { // Normal storage. Value under 75%
+                        getAccountQuotaBar()?.apply {
+                            progress = userQuota.getRelative().toInt()
+                            isVisible = true
+                            progressTintList = ColorStateList.valueOf(resources.getColor(R.color.color_accent))
                         }
+                        getAccountQuotaText()?.text = String.format(
+                            getString(R.string.drawer_quota),
+                            DisplayUtils.bytesToHumanReadable(userQuota.used, this, true),
+                            DisplayUtils.bytesToHumanReadable(userQuota.getTotal(), this, true),
+                            userQuota.getRelative()
+                        )
+                        getAccountQuotaStatusText()?.visibility = View.GONE
                     }
                 }
-                is UIResult.Loading -> getAccountQuotaText()?.text = getString(R.string.drawer_loading_quota)
-                is UIResult.Error -> getAccountQuotaText()?.text = getString(R.string.drawer_unavailable_used_storage)
```

Comment:
We're losing these 2 cases!! We shouldn't, they are so necessary

Reply:
I've replace `BaseUseCase` by `BaseUseCaseWithResult` and now, we have same cases than before